### ISSUE TITLE
Wrong error message for duplicate dependencies

### ISSUE THREAD

**Issue by s-zanella** [FStarLang/FStar#494]
```F# 
// N.fst
module N
open M
```
Suppose there are two files named `M.fst` implementing M, one in the current dir and another in `lib`
F* gives a confusing error message
```
$ fstar --include lib N.fst
Error: Found both a .fs and a .fst (or both a .fsi and a .fsti) (M.fst)
```
I think there is a use case for having several implementations or interfaces for the same module in different directories, and this shouldn't be a problem as long as there is a convention for resolving dependencies (e.g. files in an include path appearing earlier in the command-line have higher priority, interfaces have priority over implementations). Would that be too difficult to implement, too confusing for users?

**Comment by s-zanella** [comment]
It seems that this line (https://github.com/FStarLang/FStar/blob/master/src/parser/dep.fs#L91) compares `existing_file` with its full path against just the basename of `f`.

**Comment by s-zanella** [comment]
I believe this has been fixed by 5ce85419e2abf490715a0a7863f48e91dfa8a357, which establishes a precedence for resolving dependencies:

```The precedence is determined by the order of the --include
arguments passed on the command-line. If at least an
interface exists, we generate a dependency on the interface
with the highest precedence. If no interface exists, we
generate a dependency on the implementation with the highest
precedence.```

See also https://github.com/FStarLang/FStar/wiki/Dealing-with-F%E2%98%85-dependencies#when-invoking-f-in-batch-non-interactive-mode for more details.
